Announce Effective Programming in Scala and other MOOCs updates (#1231)
scala/scala-lang: blog/_posts/2021-05-04-effective-programming-scala-course.md

---
layout: blog-detail
post-type: blog
by: Julien Richard-Foy
title: "New Online Course: Effective Programming in Scala"
---

We are excited to announce that our new MOOC “Effective Programming in Scala” is now
live on Coursera! We are also happy to share that most of our existing courses have already
been updated to Scala 3.

## Effective Programming in Scala

The [Effective Programming in Scala] course

> is 6 weeks long, and aims to teach non-Scala programmers everything they need to become

Reviewer: Phrasing it like this makes it feel to me that the '6 weeks long' part is the most important. I think that should be mentioned later.

Reviewer 2: I agree on this feeling.
You could lighten with 
```suggestion
aims to teach non-Scala programmers everything they need to become professional Scala 3 programmers in six weeks.
```

Author: I agree. I have included Vince’s suggestion.

professional Scala 3 programmers. This course shows you how to embrace both functional
programming and object-oriented programming in a simple and practical way to write software
at scale. It explains how to achieve typical programming tasks in Scala,
such as modeling business domains, breaking down complex problems into simpler problems,
manipulating data, or running parallel tasks. Throughout the course, you will also learn
the best practices for writing high-quality Scala code that scales to large applications,
how to handle errors, how to write tests, and how to leverage a productive development
environment.

The course features six auto-graded programming assignments, which have been designed to
be both fun and close to real-world programs. Topics covered by the assignments include

> graphic animations, web applications, property-based testing, generic programming,

Reviewer 3: ```suggestion
graphical animations, web applications, property-based testing, generic programming,
```
or
```suggestion
graph animations, web applications, property-based testing, generic programming,
```
depending on what you want to say.

"graphic" is an adjective that really does not mean the right thing, here.

Noel Welsh has a long experience in teaching Scala, both in the industry and
within the [Scala Bridge] project.

## Other Scala Courses

The Scala Center and Martin Odersky have also been working hard to update all our
existing courses to Scala 3 (except the courses that use Spark, which still use
Scala 2.12).

This is a huge effort, and some parts are still in progress. Here is the current
status, at the time this blog is written:

- The [Scala Specialization] is a suite of five courses that teach Scala and
  data processing (possibly in parallel or distributed),
  - [Functional Programming Principles in Scala] has been updated to Scala 3
    (all the lectures and assignments have been updated),
  - [Functional Program Design] has been updated to Scala 3 (all the lectures
    and assignments have been updated),
  - [Parallel Programming] has been updated to Scala 3 (only the assignments
    have been updated),
  - [Big Data Analysis with Scala and Spark] is still on Scala 2.12,
  - [Functional Programming in Scala Capstone] is still on Scala 2.12,
- [Programming Reactive Systems] has been updated to Scala 3 (only the
  assignments have been updated).

> We will update our Spark courses to Scala 3 as soon as there is a stable release of
> Spark that supports Scala 3.

Reviewer 3: ```suggestion
We will update our Spark courses to Scala 3 as soon as there is a stable release of
Spark that supports Scala 3.
```
Perhaps mention that this may be because it supports 2.13 and that Scala 3 can use 2.13 libraries (that don't have macros)?

> Spark that supports Scala 3.

Reviewer 2: I am not sure yet if we need Scala 3 support in Spark. Scala 2.13 support could be enough
